crates/sui-benchmark/tests/simtest.rs

```diff
@@ -256,6 +262,61 @@ mod test {
         assert_eq!(expected_checkpoint, pruned);
     }
 
+    #[sim_test(config = "test_config()")]
+    async fn test_deepbook_upgrade() {
+        // make sure we can see the DeepBook package and publish dependent packages
+        let min_ver = 5; // first version with DeepBook
+        let max_ver = 5;
```

Comment:
This is not a real upgrade test because the network would start at version 5, instead of upgrading from 4 to 5, is that intended?

Reply:
I think https://github.com/MystenLabs/sui/blob/main/crates/sui/tests/protocol_version_tests.rs#L97 tries all possible upgrades. Does explicitly upgrading, then checking DeepBook-related stuff test something dfferent?

My first attempt at a test was actually trying to upgrade from 4 to 5 (by copying the approach `test_testnet_upgrade_compatibility`), but despite using the same code, the test always hung at `                test_cluster.wait_for_all_nodes_upgrade_to(version).await;`

Reply:
> I think https://github.com/MystenLabs/sui/blob/main/crates/sui/tests/protocol_version_tests.rs#L97 tries all possible upgrades. Does explicitly upgrading, then checking DeepBook-related stuff test something dfferent?

No, that test does not exercise the upgrade from 4->5.

test_testnet_upgrade_compatibility does check the upgrade from 4 to 5 is successful.

If the intention of this test is not upgrade, there is no need to specify the versions at all. Also the name of this test should be changed. I can also take a look at why the test hangs if you could show me the test code

Reply:
Thanks, I will update with the hanging code